heartbeat/powervs-subnet.in

```diff
@@ -0,0 +1,1053 @@
+
+    agent.add_parameter(
+        "ip",
+        shortdesc="IP address",
+        longdesc=(
+            "IP address within the subnet. The IP address moves together with the subnet."
+        ),
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "cidr",
+        shortdesc="CIDR",
+        longdesc="Classless Inter-Domain Routing (CIDR) of the subnet.",
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "subnet_name",
+        shortdesc="Name of the subnet",
+        longdesc="Name of the subnet. If not specified, CIDR is used as name.",
+        content_type="string",
+        required=False,
+    )
+
+    agent.add_parameter(
+        "api_type",
+        shortdesc="API type",
+        longdesc="Connect to Power Virtual Server regional endpoints over a public or private network (public|private).",
+        content_type="string",
+        required=True,
+        default="private",
+    )
+
+    agent.add_parameter(
+        "region",
+        shortdesc="Power Virtual Server region",
+        longdesc=(
+            "Region that represents the geographic area where the instance is located. "
+            "The region is used to identify the Cloud API endpoint."
+        ),
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "api_key",
+        shortdesc="API Key or @API_KEY_FILE_PATH",
+        longdesc=(
+            "API Key or @API_KEY_FILE_PATH for IBM Cloud access. "
+            "The API key content or the path of an API key file that is indicated by the @ symbol."
+        ),
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "crn_host_map",
+        shortdesc="Mapping of hostnames to IBM Cloud CRN",
+        longdesc=(
+            "Map the hostname of the Power Virtual Server instance to the CRN of the Power Virtual Server workspaces hosting the instance. "
+            "Separate hostname and CRN with a colon ':', separate different hostname and CRN pairs with a semicolon ';'. "
+            "Example: hostname01:CRN-of-Instance01;hostname02:CRN-of-Instance02"
+        ),
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "vsi_host_map",
+        shortdesc="Mapping of hostnames to PowerVS instance ids",
+        longdesc=(
+            "Map the hostname of the Power Virtual Server instance to its instance id. "
+            "Separate hostname and instance id with a colon ':', separate different hostname and instance id pairs with a semicolon ';'. "
+            "Example: hostname01:instance-id-01;hostname02:instance-id-02"
+        ),
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "proxy",
+        shortdesc="Proxy",
+        longdesc="Proxy server to access IBM Cloud API endpoints.",
+        content_type="string",
+        required=False,
```

Comment:
Use "false" and "true" for all these, as they're strings. You'll probably have to use `ocf.is_true()` for the values as well.

Reply:
Hmm, according to the function definition in ocf.py, the `required` parameter looks like a boolean and not a string:

`def add_parameter(self, name, shortdesc="", longdesc="", content_type="string", unique=False, required=False, default=None):`